File: Bookmark_sorter/cBkExtract.vb
Imports System.Text.RegularExpressions
Public Class CBookmarkExtractor
    Public Sub ExtractBookmark(ByVal vstrBookmark As String)
        Dim oFile As CFileManager
        oFile = New CFileManager
        Dim mstrBookmark As String
        mstrBookmark = MYIdentifyLinks(vstrBookmark)     
        oFile.SaveLinks(mstrBookmark)
    End Sub

    Public Function MYIdentifyLinks(ByVal htmlText As String) As String
        Dim hrefRegex As New Regex(
            "<A[^>]*?HREF\s*=\s*""([^""]+)""[^>]*?>([\s\S]*?)<\/A>",
            RegexOptions.IgnoreCase)
        Dim output As String = ""
        For Each m As Match In hrefRegex.Matches(htmlText)       '          
            output &= m.Groups(1).Value
            Return output
        Next
    End Function
End Class

File: Bookmark_sorter/cBkExtract.cs
using System.Text.RegularExpressions;
public class CBookmarkExtractor
{
	public void ExtractBookmark(string vstrBookmark)
	{
		CFileManager oFile = default(CFileManager);
		oFile = new CFileManager();
		string mstrBookmark = null;
		mstrBookmark = MYIdentifyLinks(vstrBookmark);
		oFile.SaveLinks(mstrBookmark);
	}

	public string MYIdentifyLinks(string htmlText)
	{
		Regex hrefRegex = new Regex("<A[^>]*?HREF\\s*=\\s*\"([^\"]+)\"[^>]*?>([\\s\\S]*?)<\\/A>", RegexOptions.IgnoreCase);
		string output = "";
		//          
		foreach (Match m in hrefRegex.Matches(htmlText)) {
			output += m.Groups(1).Value;
			return output;
		}
	}
}
